### DISCUSSION TITLE
JSON object requested, multiple (or no) rows returned

### DISCUSSION THREAD

**Discussion by zakaria-chahboun** [supabase/supabase#3343] [Questions] [answered]
This is a part of my code:
```js
const {data, error} = await supabase
			.from('test')
			.select('question(question_title, answer(*))')
			.eq('test_id', ID)
			.single();
```

When i select a non existent value it's gives me this error message:
```json
{
  "message": "JSON object requested, multiple (or no) rows returned",
  "details": "Results contain 0 rows, application/vnd.pgrst.object+json requires 1 row"
}
```
Whiteout any error number or just return a Null value!

What is the solution? Thanks :heart_decoration:

**Comment by zakaria-chahboun** [comment, marked as answer]
I know this is happens cuze of `.single()` behaviour. For the moment i replaced `.single()` with `.limit(1)`.

**Reply by soedirgo** [reply]
Yup, `.single()` only works if you select exactly 1 row, so you want to use it with `.limit(1)`. If you expect to get 0 rows though you might want to use `.maybeSingle()` instead.

**Reply by zakaria-chahboun** [reply]
@soedirgo Thanks!

**Comment by mike-ayres** [comment]
Unrelated answer, but I ended up with the same error message when looking for a tri-state boolean value with `.neq('field', true`)`.

Postgres expects `where field is not true` instead of `where field != true`. I think my SQL Server memory was kicking in, where null compared to anything would be false (only `is null` would work, though I didn't validate because I don't have SSMS set up on this machine).

Changing the query to `.not('field', 'is', 'true)` worked.
